core/src/test/java/org/mskcc/cbio/portal/scripts/TestIntegrationTest.java

```diff
@@ -184,7 +184,7 @@ public void testLoadStudyEs0() throws Throwable {
             geneticProfileStableIds.add("study_es_0_fusion");
```

Comment:
hi @inodb, thank you for the review. Let me see if I can explain well.
Since fusions are currently stored in the mutation_event table, fusions have to use the mutation profile in order to get the data inserted into the database, which was okay during the time when we did not have the fusion gene table in the summary view. But now, since we added the fusion gene table in the summary view, it always shows up regardless if a study has fusion data or not, because the frontend uses genetic profiles to control the visibility of gene tables (please see [issue](https://github.com/cBioPortal/cbioportal/issues/6709) reported by Tali). To fix the issue, we decided to give fusion data its own genetic profile, which is a **_STRUCTURAL_VARIANT_** genetic type profile with a **__fusion_** stableId
```
mysql> select distinct genetic_alteration_type from genetic_profile;
+-------------------------+
| genetic_alteration_type |
+-------------------------+
| COPY_NUMBER_ALTERATION  |
| STRUCTURAL_VARIANT      |
| MUTATION_EXTENDED       |
| PROTEIN_LEVEL           |
| METHYLATION             |
| MRNA_EXPRESSION         |
+-------------------------+
6 rows in set (0.00 sec)
```
```
mysql> select * from genetic_profile where stable_id like '%_fusion';
+--------------------+------------------------+-----------------+-------------------------+----------+---------+-------------+------------------------------+-----------------+------------+--------------------+
| GENETIC_PROFILE_ID | STABLE_ID              | CANCER_STUDY_ID | GENETIC_ALTERATION_TYPE | DATATYPE | NAME    | DESCRIPTION | SHOW_PROFILE_IN_ANALYSIS_TAB | PIVOT_THRESHOLD | SORT_ORDER | GENERIC_ASSAY_TYPE |
+--------------------+------------------------+-----------------+-------------------------+----------+---------+-------------+------------------------------+-----------------+------------+--------------------+
|                 45 | msk_impact_2017_fusion |              16 | STRUCTURAL_VARIANT      | FUSION   | Fusions | Fusions.    |                            1 |            NULL | NULL       | NULL               |
|                122 | breast_msk_2018_fusion |              39 | STRUCTURAL_VARIANT      | FUSION   | Fusions | Fusions.    |                            1 |            NULL | NULL       | NULL               |
+--------------------+------------------------+-----------------+-------------------------+----------+---------+-------------+------------------------------+-----------------+------------+--------------------+
2 rows in set (0.00 sec)
```

Reply:
if you take a look at the migration script we used **__sv_** STABLE_ID for fusion data (we should use _fusion instead to be consistent). The STABLE_ID does not really matter, what is important is that we give fusion data a **_GENETIC_ALTERATION_TYPE_** genetic profile. I used **__fusion_** and **_structural_variants_** STABLE_ID to distinguish old style fusion data and new structural variant data. msk-impact-2017 is the only study that has both fusions and structural variants.
```
mysql> select cs.cancer_study_identifier, gp.stable_id, gp.genetic_alteration_type, gp.datatype, gp.name from genetic_profile gp join cancer_study cs on gp.CANCER_STUDY_ID = cs.CANCER_STUDY_ID where gp.CANCER_STUDY_ID = 16 and GENETIC_ALTERATION_TYPE = 'STRUCTURAL_VARIANT';
+-------------------------+-------------------------------------+-------------------------+----------+-----------------------+
| cancer_study_identifier | stable_id                           | genetic_alteration_type | datatype | name                  |
+-------------------------+-------------------------------------+-------------------------+----------+-----------------------+
| msk_impact_2017         | msk_impact_2017_structural_variants | STRUCTURAL_VARIANT      | SV       | mskimpact2017 SV Data |
| msk_impact_2017         | msk_impact_2017_fusion              | STRUCTURAL_VARIANT      | FUSION   | Fusions               |
+-------------------------+-------------------------------------+-------------------------+----------+-----------------------+
2 rows in set (0.00 sec)
```

Reply:
good catch! changed.

Reply:
Thanks @khzhu - that explains a lot